kafka/producer.c

```diff
@@ -211,137 +211,9 @@ lua_producer_msg_delivery_poll(struct lua_State *L) {
-
-        destroy_log_msg(msg);
-
-        if (err_str != NULL) {
-            break;
-        }
-    }
-    lua_pushnumber(L, (double)count);
-    if (err_str != NULL) {
-        lua_pushstring(L, err_str);
-    } else {
-        lua_pushnil(L);
-    }
-    return 2;
-}
-
-int
-lua_producer_poll_stats(struct lua_State *L) {
-    if (lua_gettop(L) != 2)
-        luaL_error(L, "Usage: count, err = producer:poll_stats(limit)");
-
-    producer_t *producer = lua_check_producer(L, 1);
-    if (producer->event_queues == NULL ||
-            producer->event_queues->stats_queue == NULL ||
-            producer->event_queues->stats_cb_ref == LUA_REFNIL) {
-        lua_pushnumber(L, 0);
-        lua_pushliteral(L, "Consumer poll stats error: callback for logs is not set");
-        return 2;
-    }
-
-    int limit = lua_tonumber(L, 2);
-    char *json = NULL;
-    int count = 0;
-    char *err_str = NULL;
-    while (count < limit) {
-        json = queue_pop(producer->event_queues->stats_queue);
-        if (json == NULL)
-            break;
-        count++;
-        lua_rawgeti(L, LUA_REGISTRYINDEX, producer->event_queues->stats_cb_ref);
-        lua_pushstring(L, json);
-        /* do the call (1 arguments, 0 result) */
-        if (lua_pcall(L, 1, 0, 0) != 0)
-            err_str = (char*)lua_tostring(L, -1);
-
-        free(json);
-
-        if (err_str != NULL)
-            break;
-    }
-    lua_pushnumber(L, (double)count);
-    if (err_str != NULL) {
-        lua_pushstring(L, err_str);
-        return 2;
-    }
-    return 1;
-}
-
-int
-lua_producer_poll_errors(struct lua_State *L) {
-    if (lua_gettop(L) != 2)
-        luaL_error(L, "Usage: count, err = producer:poll_errors(limit)");
-
-    producer_t *producer = lua_check_producer(L, 1);
-    if (producer->event_queues == NULL || producer->event_queues->error_queue == NULL || producer->event_queues->error_cb_ref == LUA_REFNIL) {
-        lua_pushnumber(L, 0);
-        lua_pushliteral(L, "Producer poll errors error: callback for logs is not set");
-        return 2;
-    }
-
-    int limit = lua_tonumber(L, 2);
-    error_msg_t *msg = NULL;
-    int count = 0;
-    char *err_str = NULL;
-    while (count < limit) {
-        msg = queue_pop(producer->event_queues->error_queue) ;
-        if (msg == NULL) {
-            break;
-        }
-        count++;
-        lua_rawgeti(L, LUA_REGISTRYINDEX, producer->event_queues->error_cb_ref);
-        lua_pushstring(L, msg->reason);
-        /* do the call (1 arguments, 0 result) */
-        if (lua_pcall(L, 1, 0, 0) != 0) {
-            err_str = (char *)lua_tostring(L, -1);
-        }
-
-        destroy_error_msg(msg);
-
-        if (err_str != NULL) {
-            break;
-        }
-    }
-    lua_pushnumber(L, (double)count);
-    if (err_str != NULL) {
-        lua_pushstring(L, err_str);
-    } else {
-        lua_pushnil(L);
-    }
-    return 2;
-}
+LUA_RDKAFKA_POLL_FUNC(producer, poll_logs, LOG_QUEUE, destroy_log_msg, push_log_cb_args)
```

Comment:
Is it possible to use `LUA_RDKAFKA_POLL_FUNC ` for `lua_producer_msg_delivery_poll` function generation?

Reply:
delivery and consume callbacks have its own different logic. So, I didn't touch them.